File: frontend/src/types/question.ts
/** 题目类型定义 */

export interface QuestionOption {
  label: string;
  content_latex: string;
}

export interface QuestionKnowledgePoint {
  id: number;
  name: string;
  level: string;
  stage: string | null;
  grade: string | null;
}

export interface QuestionVersion {
  version_number: number;
  change_reason: string | null;
  created_at: string;
}

export interface Question {
  id: number;
  content_latex: string;
  content_plain: string;
  question_type: QuestionType;
  difficulty: number;
  score: number | null;
  answer_latex: string | null;
  solution_steps: string[] | null;
  options: QuestionOption[] | null;
  source: string | null;
  source_year: number | null;
  region: string | null;
  exam_type: string | null;
  is_ai_generated: boolean;
  review_status: ReviewStatus;
  current_version: number;
  knowledge_points: QuestionKnowledgePoint[];
  versions?: QuestionVersion[];
}

export type QuestionType = 'choice' | 'fill_blank' | 'short_answer' | 'proof' | 'comprehensive';

export type ReviewStatus = 'pending' | 'approved' | 'rejected' | 'needs_revision';

export interface QuestionCreate {
  content_latex: string;
  content_plain: string;
  question_type: QuestionType;
  difficulty: number;
  score?: number;
  answer_latex?: string;
  solution_steps?: string[];
  options?: QuestionOption[];
  source?: string;
  source_year?: number;
  region?: string;
  exam_type?: string;
  is_ai_generated?: boolean;
  review_status?: ReviewStatus;
  knowledge_point_ids?: number[];
}

export interface QuestionUpdate {
  content_latex?: string;
  content_plain?: string;
  question_type?: QuestionType;
  difficulty?: number;
  score?: number;
  answer_latex?: string;
  solution_steps?: string[];
  options?: QuestionOption[];
  source?: string;
  source_year?: number;
  region?: string;
  exam_type?: string;
  review_status?: ReviewStatus;
  change_reason?: string;
}

export interface QuestionSearchParams {
  q?: string;
  mode?: 'keyword' | 'semantic' | 'hybrid';
  stage?: string;
  grade?: string;
  question_type?: QuestionType;
  difficulty_min?: number;
  difficulty_max?: number;
  knowledge_id?: number;
  source?: string;
  review_status?: ReviewStatus;
}

export const QUESTION_TYPE_LABELS: Record<QuestionType, string> = {
  choice: '选择题',
  fill_blank: '填空题',
  short_answer: '解答题',
  proof: '证明题',
  comprehensive: '综合题',
};

export const QUESTION_TYPE_COLORS: Record<QuestionType, string> = {
  choice: 'blue',
  fill_blank: 'green',
  short_answer: 'orange',
  proof: 'purple',
  comprehensive: 'red',
};

export const REVIEW_STATUS_LABELS: Record<ReviewStatus, string> = {
  pending: '待审核',
  approved: '已通过',
  rejected: '已拒绝',
  needs_revision: '需修改',
};

File: backend/app/schemas/question.py
"""题目模式."""

from pydantic import BaseModel, Field


class QuestionBase(BaseModel):
    """题目基础字段."""

    content_latex: str
    content_plain: str
    question_type: str  # 'choice'/'fill_blank'/'short_answer'/'proof'/'comprehensive'
    difficulty: float = Field(ge=1.0, le=5.0)
    score: float | None = None
    answer_latex: str | None = None
    solution_steps: dict | None = None
    options: dict | None = None
    source: str | None = None
    source_year: int | None = None
    region: str | None = None
    exam_type: str | None = None
    is_ai_generated: bool = False
    review_status: str = "pending"


class QuestionCreate(QuestionBase):
    """创建题目."""

    knowledge_point_ids: list[int] = []


class QuestionUpdate(BaseModel):
    """更新题目."""

    content_latex: str | None = None
    content_plain: str | None = None
    question_type: str | None = None
    difficulty: float | None = Field(default=None, ge=1.0, le=5.0)
    score: float | None = None
    answer_latex: str | None = None
    solution_steps: dict | None = None
    options: dict | None = None
    source: str | None = None
    source_year: int | None = None
    region: str | None = None
    exam_type: str | None = None
    review_status: str | None = None
    change_reason: str | None = None


class QuestionRead(QuestionBase):
    """题目响应."""

    id: int
    current_version: int
    knowledge_points: list = []

    model_config = {"from_attributes": True}


class QuestionSearchParams(BaseModel):
    """题目搜索参数."""

    q: str | None = None
    mode: str = "keyword"  # 'keyword'/'semantic'/'hybrid'
    stage: str | None = None
    grade: str | None = None
    question_type: str | None = None
    difficulty_min: float | None = None
    difficulty_max: float | None = None
    knowledge_id: int | None = None
    source: str | None = None
    review_status: str | None = None


class QuestionBatchImport(BaseModel):
    """批量导入题目."""

    questions: list[QuestionCreate]
